lib/Infer/DataflowPruning.cpp

```diff
@@ -106,13 +133,41 @@ std::vector<ValueCache> DataflowPruningManager::generateInputSets(
   }
   InputSets.push_back(Cache);
 
-  Current = 2*Current + 1;
   for (auto &&I : Inputs) {
     if (I->K == souper::Inst::Var)
-      Cache[I] = {llvm::APInt(I->Width, Current++)};
+      Cache[I] = {llvm::APInt(I->Width, 1)};
+  }
+  InputSets.push_back(Cache);
+
+  for (auto &&I : Inputs) {
+    if (I->K == souper::Inst::Var)
+      Cache[I] = {llvm::APInt(I->Width, -1)};
   }
   InputSets.push_back(Cache);
 
+  for (auto &&I : Inputs) {
+    if (I->K == souper::Inst::Var)
+      Cache[I] = {llvm::APInt(I->Width, 0xFFF)};
+  }
+  InputSets.push_back(Cache);
+
+  std::srand(0);
+  for (int i = 0 ; i < 5; ++i ) {
+    for (auto &&I : Inputs) {
+      if (I->K == souper::Inst::Var)
+        Cache[I] = {llvm::APInt(I->Width, std::rand() % llvm::APInt(I->Width, -1).getLimitedValue())};
```

Comment:
this does't work... you need a story for values larger than RAND_MAX

Reply:
What do you suggest?
(Probably fine to keep this for now, and switch to C++ <random> for experiments)